wiring/inc/spark_wiring_version.h

```diff
@@ -44,8 +44,10 @@ struct __ApplicationProductVersion {
 #endif
 
 #if PLATFORM_ID!=3
-#define PRODUCT_ID(x)           __ApplicationProductID __appProductID(x); __attribute__((externally_visible, section(".modinfo.product_id"))) uint16_t __system_product_id = (x);
-#define PRODUCT_VERSION(x)       __ApplicationProductVersion __appProductVersion(x); __attribute__((externally_visible, section(".modinfo.product_version"))) uint16_t __system_product_version = (x);
+#define PRODUCT_ID(x)   _Pragma ("GCC error \"The PRODUCT_ID macro must be removed from your firmware source code.\
+        The same compiled firmware binary may be used in multiple products that share the same platform and functionality.\"")
+// Note: the following must be contained on one line
```

Comment:
Also doesn't need to be on one line :) Just use `\`

Reply:
Easier said than done, I tried and found many linker errors 🤷🏼

Reply:
Works just fine:
```c++
#define PRODUCT_VERSION(x) \
        __ApplicationProductVersion __appProductVersion(x); \
        __attribute__((externally_visible, section(".modinfo.product_version"))) \
            uint16_t __system_product_version = (x); \
        /* PRODUCT_ID used to do the following with a dynamic ID, but now we hardcode the PLATFORM_ID as the PRODUCT_ID. */ \
        __ApplicationProductID __appProductID(PLATFORM_ID); \
        __attribute__((externally_visible, section(".modinfo.product_id"))) \
            uint16_t __system_product_id = (PLATFORM_ID);
```